core/src/verified_vote_packets.rs

```diff
@@ -213,90 +213,89 @@ impl VerifiedVotePackets {
         let vote_packets = vote_packets_receiver.recv_timeout(RECV_TIMEOUT)?;
         let vote_packets = std::iter::once(vote_packets).chain(vote_packets_receiver.try_iter());
 
+        if !would_be_leader {
+            return Ok(());
+        }
```

Comment:
I think the point of needing to drain the channel is subtle enough that a comment would be good:
```suggestion
        // No need to process any votes if we will not be the leader soon. But,
        // return early only after draining the channel to avoid accumulating
        // votes that will be stale by the time we do become leader
        if !would_be_leader {
            return Ok(());
        }
```

Reply:
that make sense, commited